docs/docs/typescript/generated-types.md

```diff
@@ -0,0 +1,157 @@
+  return db.post.findMany()
+}
+
+// highlight-next-line
+export const post: QueryResolvers['post'] = ({ id }) => {
+  return db.post.findUnique({
+    where: { id },
+  })
+}
+```
+
+These types help you by making sure you're returning an object in the shape of what you've defined in your SDL. Note that these types expect you to return the _complete_ type that you've defined in your SDL. You can just return the result of the Prisma query, and not have to worry about how, for example, a DateTime in Prisma maps to a String in GraphQL.
+
+:::note A note on union types
+
+Lets say that in one of your SDLs, you define a union type
+
+```graphql
+type OutOfStock {
+  message: String!
+}
+
+// highlight-next-line
+union CandyResult = Candy | OutOfStock
+
+type Query {
+  candy(id: String!): CandyResult @skipAuth
+}
+```
+
+These types will also be handled automatically. But if you're returning a different Prisma model, you may need to write your own resolver type, as the type generator won't know how to map the Prisma type to the GraphQL return type.
+
+:::
+
+## Under the Hood
+
+Redwood uses [GraphQL Code Generator](https://www.graphql-code-generator.com) (aka graphql-codegen) to generate types for your GraphQL operations and SDLs. It's even configured to use the types from your generated Prisma Client, to make sure that your resolvers are strongly typed!
+
+### Customizing GraphQL Code Generation
+
+While the default settings are configured so that things just work️, you can customize them to your liking by adding a `./codegen.yml` file to the root of your project.
+
+:::info Curious about the defaults?
+
+You can find them [here](https://github.com/redwoodjs/redwood/blob/main/packages/internal/src/generate/graphqlCodeGen.ts) in Redwood's source. Look for the `generateTypeDefGraphQLWeb` and `generateTypeDefGraphQLApi` functions.
+
+:::
+
+For example, adding this `codegen.yml` to the root of your project will transform the names of the generated types to UPPERCASE:
+
+```yml title="codegen.yml"
+config:
+  namingConvention:
+    typeNames: change-case-all#upperCase
+```
+
+You can configure graphql-codegen in a number of different ways: `codegen.yml`, `codegen.json`, or `codegen.js`. Even a `codegen` key in your root `package.json` will do. graphql-codegen uses [cosmiconfig](https://github.com/davidtheclark/cosmiconfig#cosmiconfig) under the hood—take a look at their docs if you want to know more.
+
+For completeness, [here's the docs](https://www.graphql-code-generator.com/docs/config-reference/config-field) on configuring GraphQL Code Generator. Currently, Redwood only supports the root level `config` option.
+
+:::tip Using VSCode?
+
+As a part of type generation, the [VSCode GraphQL extension](https://marketplace.visualstudio.com/items?itemName=GraphQL.vscode-graphql) configures itself based on the merged schema Redwood generates in `.redwood/schema.graphql`.
+You can configure it further in `graphql.config.js` at the root of your project.
```

Comment:
This is only related cause the extensions consumes the generated types right?

Reply:
Ummm the extension consumes the schema, but if you configure the graphql.config.js file, you can also tell the extension to include certain headers (e.g. for auth) and make queries right in vscode